Controller/Process/Json.php

```diff
@@ -432,4 +450,26 @@ protected function _returnResult($message)
             ->setBody($message);
         return;
     }
+
+    /**
+     * Returns true when the event code support HMAC validation
+     *
+     * @param $response
+     */
+    protected function _isHmacSupportedEventCode($response)
+    {
+        $eventCodes = array("ADVICE_OF_DEBIT","AUTHORISATION","AUTHORISATION_PENDING","AUTHORISE_REFERRAL","CANCELLATION",
+            "CANCEL_OR_REFUND", "CAPTURE","CAPTURE_FAILED","CAPTURE_WITH_EXTERNAL_AUTH","CHARGEBACK","CHARGEBACK_REVERSED",
+            "DEACTIVATE_RECURRING","FRAUD_ONLY","FUND_TRANSFER","HANDLED_EXTERNALLY","MANUAL_REVIEW_ACCEPT","NOTIFICATION_OF_CHARGEBACK",
+            "NOTIFICATION_OF_FRAUD", "OFFER_CLOSED","ORDER_OPENED","PAIDOUT_REVERSED","PAYOUT_DECLINE","PAYOUT_EXPIRE","PAYOUT_THIRDPARTY",
+             "PREARBITRATION_LOST","PREARBITRATION_WON","PROCESS_RETRY","RECURRING_CONTRACT","REFUND","REFUNDED_REVERSED","REFUND_FAILED",
+            "REFUND_WITH_DATA", "REQUEST_FOR_INFORMATION","SECOND_CHARGEBACK","SUBMIT_RECURRING","VOID_PENDING_REFUND",
+            "POSTPONED_REFUND","TECHNICAL_CANCEL","AUTHORISATION_ADJUSTMENT","CANCEL_AUTORESCUE","AUTORESCUE");
+        if (array_key_exists('eventCode', $response)) {
+            if (in_array($response['eventCode'], $eventCodes)) {
```

Comment:
we can merge this into on if statement. in_array is enough we don't need the array_key_exists method.